### PR TITLE
Check and apply max deduction for handgrading annotations

### LINKED ISSUE #502: Local score update when applying annotations past the max deduction is incorrect
If an annotation that has a max deduction set is applied multiple times, the score on the page keeps going down even though the value on the server is correct.

### PR DESCRIPTION
**Pull request by broad-well**
This PR fixes #502. When an annotation is applied or deleted in the Handgrading interface, we compute the effective change to the score based on its max deduction and total number of applications so far.

Tests are included for both applying and deleting annotations across the max deduction boundary. Additional tests are included for cases where the max deduction of an annotation is not a multiple of its unit deduction, such as ("-3/-7 max"). This PR's implementation changes makes those tests pass.

This PR is a copy of #512, which had to be reopened to circumvent GitHub's [erroneous](https://github.com/orgs/community/discussions/58535#discussioncomment-6681877) stale review checks.

### CODE REVIEW

File: tests/test_components/test_project_view/test_submission_detail/test_ag_test_command_result_detail.ts [hunk 1]
```diff
@@ -164,17 +164,6 @@ describe('Correctness feedback tests', () => {
             );
         });
 
-        test('Actual return code available, return code not checked', async () => {
```
Comment (james-perretta) [inline tests/test_components/test_project_view/test_submission_detail/test_ag_test_command_result_detail.ts]: This looks like the result of some merge weirdness (this was a test from the other PR, right?), can you remove this change from the pr?

**Review by james-perretta — CHANGES_REQUESTED** [review]

File: tests/test_components/test_project_view/test_submission_detail/test_ag_test_command_result_detail.ts (hunk 1 above)
Comment (Normanwqn) [reply tests/test_components/test_project_view/test_submission_detail/test_ag_test_command_result_detail.ts]: I used `git rebase -i` to drop that commit from the PR :). Please let us know soon if this works :)

File: src/components/project_view/submission_detail/ag_test_command_result_detail.vue [hunk 2]
```diff
@@ -33,6 +33,8 @@
                         {'actual-return-code-correct': ag_test_command_result !== null
                           && ag_test_command_result.return_code_correct,
                           'actual-return-code-incorrect': ag_test_command_result !== null
+                          // return_code_correct can be null so we need to check for false
```
Comment (james-perretta) [inline src/components/project_view/submission_detail/ag_test_command_result_detail.vue]: Hmm why is it showing up as a new change now? I see this comment on develop after pulling locally...

Comment (Normanwqn) [reply src/components/project_view/submission_detail/ag_test_command_result_detail.vue]: Oops. I think this also came from the other PR. I have now deleted the comment commit from this branch :)

**Review by james-perretta — APPROVED** [review]

**Comment by james-perretta** [comment]
Thanks again for this!

**Pull request merged**